core/transaction/entity.go

```diff
@@ -386,3 +390,56 @@ func VerifyTransaction(txnHash string, sharders []string) (*Transaction, error)
 	return nil, errors.Throw(ErrTooLessConfirmation, strings.Join(msgList, "\r\n"))
 
 }
+
+// SuggestTransactionFeeFromMiners estimates transaction fee, returns next round off fee (if received float from miner)
+func (t *Transaction) SuggestTransactionFeeFromMiners(miners []string, numMinersToCheck int) (float64, error) {
+	// average out random miners result (prevent trust from rogue miners
+	randomMiners := util.GetRandom(miners, numMinersToCheck)
+
+	result := make(chan *util.PostResponse, len(randomMiners))
+	defer close(result)
+
+	for _, miner := range randomMiners {
+		go func(minerurl string) {
+			url := minerurl + ESTIMATE_TRANSACTION_COST
+			req, err := util.NewHTTPPostRequest(url, t)
+			if err != nil {
+				zap.S().Error(minerurl, " failed to make request. ", err.Error())
+				return
+			}
+			res, err := req.Post()
+			if err != nil {
+				zap.S().Error(minerurl, " estimate transaction cost error. ", err.Error())
+			}
+			result <- res
+		}(miner)
+	}
+
+	cumCost, costCnt := 0.0, 0
+
+	for range randomMiners {
+		rsp := <-result
```

Comment:
We will stuck here if none of the miners give response. See code above in line 408.

Reply:
Ah, yes. Even for partial responses (one fails out of 2).

Nice catch. Fixing

Reply:
Yes, resolved. PTAL - this should be fine now. Thanks